DateTimeKit/DateTime.swift

```diff
@@ -207,23 +202,89 @@ public struct DateTime {
 	
 	Also available by the `-` operator.
 	
-	:param: period The period to be subtracted
-	:returns: A new `DateTime` that represents the new dateTime
+	- parameter period: The period to be subtracted
+	- returns: A new `DateTime` that represents the new dateTime
 	*/
 	public func minus(period: Period) -> DateTime {
 		return DateTime(LocalDateTime(self.dateTime.date - period, self.dateTime.time), self.zone)
 	}
+	
+	// MARK: - Extensions added re: joda-time API (intermediate changes, more to come)
+	
+	public func withYear(year: Int) -> DateTime {
+		if let d = DateTime(year, month, day, hour, minute, second, millisecond, zone, nil) {
+			return d
+		} else {
+			return self
+		}
+	}
+	
+	public func withMonth(month: Int) -> DateTime {
+		if let d = DateTime(year, month, day, hour, minute, second, millisecond, zone, nil) {
+			return d
+		} else {
+			return self
+		}
+	}
+	
+	public func withDay(day: Int) -> DateTime {
+		if let d = DateTime(year, month, day, hour, minute, second, millisecond, zone, nil) {
+			return d
+		} else {
+			return self
+		}
+	}
+	
+	public func withHour(hour: Int) -> DateTime {
+		if let d = DateTime(year, month, day, hour, minute, second, millisecond, zone, nil) {
+			return d
+		} else {
+			return self
+		}
+	}
+	
+	public func withMinute(minute: Int) -> DateTime {
+		if let d = DateTime(year, month, day, hour, minute, second, millisecond, zone, nil) {
+			return d
+		} else {
+			return self
+		}
+	}
+	
+	public func withSecond(second: Int) -> DateTime {
+		if let d = DateTime(year, month, day, hour, minute, second, millisecond, zone, nil) {
+			return d
+		} else {
+			return self
+		}
+	}
+	
+	public func withMillisecond(millisecond: Int) -> DateTime {
+		if let d = DateTime(year, month, day, hour, minute, second, millisecond, zone, nil) {
+			return d
+		} else {
+			return self
+		}
+	}
+	
+	public func withZone(zone: Zone) -> DateTime {
+		if let d = DateTime(year, month, day, hour, minute, second, millisecond, zone, nil) {
+			return d
+		} else {
+			return self
+		}
+	}
```

Comment:
Thanks for the additional functions. I'd prefer if you used `self.` for all of the instance variables, though, so it is much clearer which variable is shadowed.